### ISSUE TITLE
Session reset by devise with use of paper_trail

### ISSUE THREAD

**Issue by bricesanchez** [rikkipitt/rails_admin_history_rollback#8]
Hello ! 

Thanks for your awesome work on `rails_admin_history_rollback` !

I've probably notice an error dependant to the `paper_trail` project.

I've use `rails_admin` (github master), `rails_admin_history_rollback` (0.0.6), `devise` (3.5.2) and `paper_trail` (~> 4.0.0)

Login is always reseted by `devise` because `paper_trail` try to access to the `current_user` before `devise` (see this for more infos : https://github.com/airblade/paper_trail/pull/556 )

My fix was to use the `paper_trail` github master source and manually add `set_paper_trail_whodunnit` before_action as the readme said (https://github.com/airblade/paper_trail#finding-out-who-was-responsible-for-a-change) : 

``` ruby
# app/controllers/application_controller.rb
class ApplicationController < ActionController::Base
  [...]
  before_action :set_paper_trail_whodunnit
  [...]
end
```

Now, it works very well.

c.c. @sferik

**Comment by rikkipitt** [comment]
@bricesanchez thanks for reporting this. I've not encountered this problem yet strangely. I'll see if I can recreate the issue and include a note in the `README.md` with reference to this issue to help future users.

**Comment by rikkipitt** [comment]
I've just been looking at the changelog over at `paper_trail` and it seem this breaking change has yet to be released as version 5.0.0 https://github.com/airblade/paper_trail/blob/master/CHANGELOG.md

@bricesanchez can you confirm which version of `paper_trail` you are using in your project? I assume if you're using rubygems it's v4.0.0 https://github.com/airblade/paper_trail/releases/tag/v4.0.0 - which at the time of writing, is the latest...

**Comment by bricesanchez** [comment]
@rikkipitt, i use the latest commit on master : https://github.com/airblade/paper_trail/commit/84bedae4bcd43ddb5162d947c5f7925dda872636 

``` ruby
# Gemfile.lock
GIT
  remote: git://github.com/airblade/paper_trail.git
  revision: 84bedae4bcd43ddb5162d947c5f7925dda872636
  branch: master
  specs:
    paper_trail (4.0.0)
      activerecord (>= 3.0, < 6.0)
      activesupport (>= 3.0, < 6.0)
      request_store (~> 1.1)
```

But it's knowned as 4.0.0 https://github.com/airblade/paper_trail/blob/master/lib/paper_trail/version_number.rb

**Comment by rikkipitt** [comment]
Ah you see, it's not really v4.0.0 when you use the master branch. You're getting all the features yet to be released. I can mention this in the `README.md` though to help others if they encounter this...

**Comment by rikkipitt** [comment]
Closed via 2149498
